typing/typedtree.mli

```diff
@@ -601,7 +601,8 @@ and omitted_parameter =
   { mode_closure : Mode.Alloc.r;
     mode_arg : Mode.Alloc.l;
     mode_ret : Mode.Alloc.l;
-    sort_arg : Jkind.sort }
+    sort_arg : Jkind.sort;
+    sort_eta_expansion_ret : Jkind.sort }
```

Comment:
I thing that this can be called just `sort_ret` because I think that `mode_ret` is the return mode for the same closure that this is the return sort of. Having them have very different names is a bit confusing.